src/dex/cables/cables.ts

```diff
@@ -0,0 +1,1006 @@
+  ) {
+    let result = [];
+
+    for (let i = 0; i < amounts.length; i++) {
+      let amt = amounts[i];
+      if (amt === 0n) {
+        result.push(amt);
+        continue;
+      }
+
+      let decimals = baseToken.decimals;
+      let out_decimals = quoteToken.decimals;
+
+      let price = this.calculatePriceSwap(
+        orderbook,
+        Number(amt) / 10 ** decimals,
+        isInputQuote,
+      );
+      result.push(BigInt(Math.round(price * 10 ** out_decimals)));
+    }
+    return result;
+  }
+
+  calculatePriceSwap(
+    prices: string[][],
+    requiredQty: number,
+    qtyMode: Boolean,
+  ) {
+    let sumBaseQty = 0;
+    let sumQuoteQty = 0;
+    const selectedRows: string[][] = [];
+
+    const isBase = qtyMode;
+    const isQuote = !qtyMode;
+
+    for (const [price, volume] of prices) {
+      if (isBase) {
+        if (sumBaseQty >= requiredQty) {
+          break;
+        }
+      }
+
+      if (isQuote) {
+        if (sumQuoteQty >= requiredQty) {
+          break;
+        }
+      }
+
+      let currentBaseQty = Number(volume);
+      let currentQuoteQty = Number(volume) * Number(price);
+
+      const overQty = isBase
+        ? currentBaseQty + sumBaseQty > requiredQty
+        : currentQuoteQty + sumQuoteQty > requiredQty;
+
+      if (overQty) {
+        if (isBase) {
+          currentBaseQty = requiredQty - sumBaseQty;
+          currentQuoteQty = currentBaseQty * Number(price);
+        }
+
+        if (isQuote) {
+          currentQuoteQty = requiredQty - sumQuoteQty;
+          currentBaseQty =
+            currentQuoteQty *
+            new BigNumber(1).dividedBy(new BigNumber(price)).toNumber();
+        }
+      }
+
+      sumBaseQty += currentBaseQty;
+      sumQuoteQty += currentQuoteQty;
+      selectedRows.push([price, currentBaseQty.toString()]);
+    }
+
+    const vSumBase = selectedRows.reduce((sum: number, [price, volume]) => {
+      return sum + Number(price) * Number(volume);
+    }, 0);
+
+    const price = new BigNumber(vSumBase)
+      .dividedBy(new BigNumber(sumBaseQty))
+      .toNumber();
+
+    if (isBase) {
+      return requiredQty / price;
+    } else {
+      return requiredQty * price;
+    }
+  }
+
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+    transferFees?: TransferFeeParams,
+    isFirstSwap?: boolean,
+  ): Promise<ExchangePrices<CablesData> | null> {
+    const isRestricted = await this.isRestricted();
+    if (isRestricted) {
+      return null;
+    }
+
+    await this.setTokensMap();
+
+    try {
+      const normalizedSrcToken = this.normalizeToken(srcToken);
+      const normalizedDestToken = this.normalizeToken(destToken);
+      // If: same token, return null
+      if (
+        normalizedSrcToken.address.toLowerCase() ===
+        normalizedDestToken.address.toLowerCase()
+      ) {
+        return null;
+      }
+
```

Comment:
Let's do this early return before any Redis calls

Reply:
fixed